### ISSUE TITLE
Validator - Device with error incorrectly marked as "correct_device"

### ISSUE THREAD

**Issue by noursaidi** [faucetsdn/udmi#403]
In validator, a devices with an error are marked as `clean_devices` and not an `error_device`

```
{
  "timestamp" : "2022-08-03T11:23:10Z",
  "version" : "1.3.14",
  "errors" : [ ],
  "summary" : {
    "correct_devices" : [ "AHU-1" ],
    "extra_devices" : [ ],
    "missing_devices" : [ "AHU-22", "GAT-123", "SNS-4" ],
    "error_devices" : [ ]
  },
  "devices" : { }
}

```

Using Pubber as the device with the following run command - `bin/pubber sites/udmi_site_model PROJECT_ID AHU-1 123 noHardware extraPoint=extra_point extraField=extra_field `

```
Processing device #1/4: AHU-1/event_system
Warning: Nashorn engine is planned to be removed from a future JDK release
Validation complete AHU-1/event_system
Processing device #1/4: AHU-1/state_system
Error validating schema: While converting to json node: 1 schema violations found
Processing device #1/4: AHU-1/state_pointset
Error validating contents: Metadata validation failed: Extra points: extra_point
Processing device #1/4: AHU-1/state
Error validating schema: While converting to json node: 1 schema violations found
Processing device #1/4: AHU-1/event_pointset
Error validating schema: While converting to json node: 1 schema violations found
Error validating contents: Metadata validation failed: Extra points: extra_point
```

**Comment by noursaidi** [comment]
Actually, when the device hen a device only has a metadata validation error, e.g. `bin/pubber sites/udmi_site_model PROJECT_ID AHU-1 123 extraPoint=extra_point` it does appear in the `error_devices` list, 

```
{
  "timestamp" : "2022-08-03T11:43:54Z",
  "version" : "1.3.14",
  "errors" : [ ],
  "summary" : {
    "correct_devices" : [ ],
    "extra_devices" : [ ],
    "missing_devices" : [ "AHU-22", "GAT-123", "SNS-4" ],
    "error_devices" : [ "AHU-1" ]
  },
  "devices" : {
    "AHU-1" : {
      "status" : {
        "message" : "Metadata validation failed: Extra points: extra_point",
        "category" : "validation.error.simple",
        "level" : 500
      }
    }
  }
}
```

But if the device only has a singular schema error (`bin/pubber sites/udmi_site_model PROJECT_ID AHU-1 123 noHardware`) it does not get marked as an `error_device`
```
{
  "timestamp" : "2022-08-03T11:48:11Z",
  "version" : "1.3.14",
  "errors" : [ ],
  "summary" : {
    "correct_devices" : [ "AHU-1" ],
    "extra_devices" : [ ],
    "missing_devices" : [ "AHU-22", "GAT-123", "SNS-4" ],
    "error_devices" : [ ]
  },
  "devices" : { }
}
```

**Comment by noursaidi** [comment]
Fixed by #439
